debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnectorTask.java

```diff
@@ -71,10 +70,10 @@ public ChangeEventSourceCoordinator<PostgresPartition, PostgresOffsetContext> st
             throw new ConnectException("Unable to load snapshotter, if using custom snapshot mode, double check your settings");
         }
 
-        heartbeatConnection = new PostgresConnection(connectorConfig.getJdbcConfig(), PostgresConnection.CONNECTION_HEARTBEAT);
-        final Charset databaseCharset = heartbeatConnection.getDatabaseCharset();
-        // The connection might be unnecessary as heartbeat need not to be enabled
-        heartbeatConnection.close();
+        final Charset databaseCharset;
+        try (PostgresConnection tempConnection = new PostgresConnection(connectorConfig.getJdbcConfig(), PostgresConnection.CONNECTION_GENERAL)) {
```

Comment:
@Naros

Reply:
Hi @jpechane is there something specific you wanted changed here?  I see you merged this, which is why I ask.

Reply:
@Naros I am sorry probably a nice or misclick